crates/flipperzero/src/furi/dialog.rs

```diff
@@ -0,0 +1,179 @@
+//! Furi dialogs.
+
+#[cfg(feature = "alloc")]
+use alloc::ffi::CString;
+
+use core::ffi::{c_char, CStr};
+use core::marker::PhantomData;
+use core::ptr;
+
+use flipperzero_sys as sys;
+use flipperzero_sys::furi::UnsafeRecord;
+
+use super::canvas::Align;
+
+const RECORD_DIALOGS: *const c_char = sys::c_string!("dialogs");
+
+#[cfg(feature = "alloc")]
+const BUTTON_OK: &'static CStr = unsafe { CStr::from_bytes_with_nul_unchecked(b"OK\0") };
+
+/// A handle to the Dialogs app.
+pub struct DialogsApp {
+    data: UnsafeRecord<sys::DialogsApp>,
+}
+
+/// A dialog message.
+pub struct DialogMessage<'a> {
+    data: *mut sys::DialogMessage,
+    _phantom: PhantomData<&'a CStr>,
+}
+
+/// Button pressed on a dialog.
+pub enum DialogMessageButton {
+    Back,
+    Left,
+    Right,
+    Center,
+}
+
+impl DialogsApp {
+    /// Obtains a handle to the Dialogs app.
+    pub fn open() -> Self {
+        Self {
+            data: unsafe { UnsafeRecord::open(RECORD_DIALOGS) },
+        }
+    }
+
+    /// Displays a message.
+    pub fn show(&mut self, message: &DialogMessage) -> DialogMessageButton {
+        let button_sys = unsafe { sys::dialog_message_show(self.data.as_ptr(), message.data) };
+
+        DialogMessageButton::from_sys(button_sys).expect("Invalid button")
+    }
+}
+
+impl<'a> DialogMessage<'a> {
+    /// Allocates a new dialog message.
+    pub fn new() -> Self {
+        let data = unsafe { sys::dialog_message_alloc() };
+        assert!(!data.is_null());
+
+        Self {
+            data,
+            _phantom: PhantomData,
+        }
+    }
+
+    /// Sets the labels of the buttons.
+    pub fn set_buttons_c(
```

Comment:
I think we can drop the `_c` suffix (it took me a moment to realize what it meant). I don't think it's unreasonable to require a `CStr` if that's what the underlying API requires (even if `CStr`/`CString` are a bit of a pain to work with).